apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ErrorReporting/ErrorReportingContextExceptionLogger.cs

```diff
@@ -158,14 +170,18 @@ private LogEntry CreateReportRequest(Exception exception, IContextWrapper contex
                 }
             };
 
-            return new LogEntry
+            var entry = new LogEntry
             {
                 Resource = _options.EventTarget.MonitoredResource,
                 LogName = _logName,
                 Severity = LogSeverity.Error,
                 Timestamp = timestamp,
                 JsonPayload = errorEvent
             };
+
+            entry.SetTraceAndSpanIfAny(_traceTarget, _serviceProvider, null);
```

Comment:
If you made the extension method return the entry, you could have this as:

```csharp
return new LogEntry
{
    Resource = _options.EventTarget.MonitoredResource,
    LogName = _logName,
    Severity = LogSeverity.Error,
    Timestamp = timestamp,
    JsonPayload = errorEvent
}.SetTraceAndSpanIfAny(_traceTarget, _serviceProvider, null);
```

Just a thought - not particularly bothered either way.

Reply:
Ah yes, will do that.